### PR TITLE
ci: use `ubuntu-latest-m` runner

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
Use large runner to fix space/memory issues in tests/releases.

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by simar7** [comment]
amazing!

File: .github/workflows/reusable-release.yaml
Comment (DmitriyLewen) [inline .github/workflows/reusable-release.yaml:1]: This workflow doesn't require many resources.
But we need to use the same runners as `test.yaml`, so as not to make different `require status checks`.

**Review by knqyf263 — APPROVED** [review]

**Comment by DmitriyLewen** [comment]
@aqua-bot backport release/v0.52

**Comment by DmitriyLewen** [comment]
@aqua-bot backport release/v0.52

**Pull request merged**